src/emit/solana.rs

```diff
@@ -2367,104 +2387,147 @@ impl<'a> TargetRuntime<'a> for SolanaTarget {
                 "offset_ptr",
             );
 
-            let offset = binary
-                .builder
-                .build_load(offset_ptr, "offset")
-                .into_int_value();
+            let new_string_length = binary.vector_len(val);
```

Comment:
I think we can use better naming. `val` sounds like the value to be saved on storage.

Reply:
But `val` *is* the value being saved to storage. I am all for better names, but I don't know what else to call this. Let me know if you have a suggestion.